File: CSharp/FetchInfoOnSportsPlayer.cs
namespace CSharp
{
    // Create a class that takes the following four arguments for a particular football player:
    // name, age, height, weight
    // Also, create three functions for the class that returns the following strings:
    // GetAge() returns "name is age age"
    // GetHeight() returns "name is height cm"
    // GetWeight() returns "name weighs weight kg"
    // https://edabit.com/challenge/ZngT4zDckDugt2JGY

    public class Player
    {
        public string Name { get; }
        public byte Age { get; }
        public byte Height { get; }
        public byte Weight { get; }

        public Player(string name, byte age, byte height, byte weight)
        {
            Name = name;
            Age = age;
            Height = height;
            Weight = weight;
        }
    }

    public class FetchInfoOnSportsPlayer
    {
        public Player Player { get; }

        public FetchInfoOnSportsPlayer(Player player)
        {
            Player = player;
        }

        public string GetAge() => $"{this.Player.Name} is age {this.Player.Age}";

        public string GetHeight() => $"{this.Player.Name} is {this.Player.Height}cm";

        public string GetWeight() => $"{this.Player.Name} weighs {this.Player.Weight}kg";
    }
}

File: js/fetchInfoOnSportsPlayer.js
// Create a class that takes the following four arguments for a particular football player:
// name, age, height, weight
// Also, create three functions for the class that returns the following strings:
// getAge() returns "name is age age"
// getHeight() returns "name is height cm"
// getWeight() returns "name weighs weight kg"
// https://edabit.com/challenge/ZngT4zDckDugt2JGY

class Player {
  constructor(name, age, height, weight) {
    this.name = name;
    this.age = age;
    this.height = height;
    this.weight = weight;
  }

  getAge() {
    return `${this.name} is age ${this.age}`;
  }

  getHeight() {
    return `${this.name} is ${this.height}cm`;
  }

  getWeight() {
    return `${this.name} weighs ${this.weight}kg`;
  }
}

module.exports = Player;
